api/src/main/java/org/apache/cloudstack/api/BaseCmd.java

```diff
@@ -456,4 +459,18 @@ public ApiCommandResourceType getApiResourceType() {
         return ApiCommandResourceType.None;
     }
 
+    public Map<String, String> convertDetailsToMap(Map details) {
+        Map<String, String> detailsMap = new HashMap<String, String>();
+        if (MapUtils.isNotEmpty(details)) {
+            Collection parameterCollection = details.values();
+            Iterator iter = parameterCollection.iterator();
+            while (iter.hasNext()) {
+                HashMap<String, String> value = (HashMap<String, String>)iter.next();
+                for (Map.Entry<String,String> entry: value.entrySet()) {
+                    detailsMap.put(entry.getKey(),entry.getValue());
+                }
+            }
+        }
+        return detailsMap;
+    }
```

Comment:
Good refactor! This is happening so often. Did you replace all occurences?

Reply:
updated 2 more APIs